source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidNearbyConnectionClient.cs

```diff
@@ -0,0 +1,399 @@
+            {
+                mClient = nearbyClass.CallStatic<AndroidJavaObject>("getConnectionsClient",
+                    AndroidHelperFragment.GetActivity());
+            }
+        }
+
+        public int MaxUnreliableMessagePayloadLength()
+        {
+            return NearbyConnectionConfiguration.MaxUnreliableMessagePayloadLength;
+        }
+
+        public int MaxReliableMessagePayloadLength()
+        {
+            return NearbyConnectionConfiguration.MaxReliableMessagePayloadLength;
+        }
+
+        public void SendReliable(List<string> recipientEndpointIds, byte[] payload)
+        {
+            InternalSend(recipientEndpointIds, payload);
+        }
+
+        public void SendUnreliable(List<string> recipientEndpointIds, byte[] payload)
+        {
+            InternalSend(recipientEndpointIds, payload);
+        }
+
+        private void InternalSend(List<string> recipientEndpointIds, byte[] payload)
+        {
+            Misc.CheckNotNull(recipientEndpointIds);
+            Misc.CheckNotNull(payload);
+
+            using (var payloadClass = new AndroidJavaClass("com.google.android.gms.nearby.connection.Payload"))
+            using (var payloadObject = payloadClass.CallStatic<AndroidJavaObject>("fromBytes", payload))
+            using (var task = mClient.Call<AndroidJavaObject>("sendPayload",
+                AndroidJavaConverter.ToJavaStringList(recipientEndpointIds),
+                payloadObject))
+                ;
+        }
+
+        public void StartAdvertising(string name, List<string> appIdentifiers,
+            TimeSpan? advertisingDuration, Action<AdvertisingResult> resultCallback,
+            Action<ConnectionRequest> connectionRequestCallback)
+        {
+            Misc.CheckNotNull(resultCallback, "resultCallback");
+            Misc.CheckNotNull(connectionRequestCallback, "connectionRequestCallback");
+
+            if (advertisingDuration.HasValue && advertisingDuration.Value.Ticks < 0)
+            {
+                throw new InvalidOperationException("advertisingDuration must be positive");
+            }
+
+            connectionRequestCallback = ToOnGameThread(connectionRequestCallback);
+            resultCallback = ToOnGameThread(resultCallback);
+
+            AdvertisingConnectionLifecycleCallbackProxy callbackProxy =
+                new AdvertisingConnectionLifecycleCallbackProxy(resultCallback, connectionRequestCallback, this);
+            using (var connectionLifecycleCallback =
+                new AndroidJavaObject("com.google.games.bridge.ConnectionLifecycleCallbackProxy", callbackProxy))
+            using (var advertisingOptions = CreateAdvertisingOptions())
+            using (var task = mClient.Call<AndroidJavaObject>("startAdvertising", name, GetServiceId(),
+                connectionLifecycleCallback, advertisingOptions))
+            {
+                AndroidTaskUtils.AddOnFailureListener(
```

Comment:
Why it is on failure?
What will happen if we launch multiple advertising at the same time?

Reply:
1. It should be onSuccessListener. Updated it. 
2. Already advertising and already discovering cases are handled in callback.